sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryAutoConfiguration.java

```diff
@@ -103,6 +103,9 @@
 
       options.setSentryClientName(BuildConfig.SENTRY_SPRING_BOOT_SDK_NAME);
       options.setSdkVersion(createSdkVersion(options));
+      if (options.getTracesSampleRate() == null) {
+        options.setTracesSampleRate(0.0);
+      }
```

Comment:
why do we need that here? isn't users choice to set it up?

Reply:
Spring has already a property "enable-tracing" which triggers auto configuration. If we don't do this change, users would have to set both `enable-tracing=true` and `traces-sample-rate=0.0` to enable integration.

Reply:
but that's what is described here right https://docs.sentry.io/platforms/java/guides/spring-boot/performance/#enable-tracing 
if it's a matter of not doing multiple things, I'd say that removing `enable-tracing` would be the way to go instead.

Reply:
We can do it but it will be breaking. Probably better to do it in another PR and leave this one as it is?

Reply:
wfm

Reply:
As decided on Friday, I'll deprecate `enable-tracing` and make `traces-sample-rate` turn on integration too.

Reply:
@maciejwalkowiak we need docs for this

Reply:
Sure, I'll prepare a PR

Reply:
after deprecating `enable-tracing`, we should remove this right now, right?

Reply:
I don't think so. This keeps the old behavior - users only need to set `enable-tracing=true` to get tracing working. If we remove it, it is a breaking change.